packages/cli/src/cli/basemaps/tileset.updater.ts

```diff
@@ -0,0 +1,239 @@
+import { Epsg } from '@basemaps/geo';
+import {
+    Aws,
+    DefaultBackground,
+    RecordPrefix,
+    TileMetadataImageRule,
+    TileMetadataImageryRecord,
+    TileMetadataNamedTag,
+    TileMetadataSetRecord,
+    TileMetadataTable,
+    TileMetadataTag,
+} from '@basemaps/shared';
+import { deepStrictEqual } from 'assert';
+import { ulid } from 'ulid';
+import { addDefaults, assertTileSetConfig, FullImageryConfig, ProjectionConfig } from './tileset.config';
+import { parseRgba, primeImageryCache, rgbaToHex } from './tileset.util';
+
+export interface TagChanges {
+    name: string;
+    projection: Epsg;
+    before: TileMetadataSetRecord;
+    after: TileMetadataSetRecord;
+}
+
+/**
+ * All the changes made by applying the TileSetConfig
+ */
+export interface TilesetChanges {
+    changes: TagChanges | null;
+    /** A mapping from an id to an imagery record */
+    imagery: Map<string, TileMetadataImageryRecord>;
+}
+
+function objectsDiffer(a: any, b: any): boolean {
+    try {
+        deepStrictEqual(a, b);
+        return false;
+    } catch (err) {}
+    return true;
+}
+
+function initiaTs(name: string, projection: Epsg): TileMetadataSetRecord {
```

Comment:
`initial`?

Reply:
This should really live inside the TileSet database class as we use something similar somewhere else?